src/server.py

```diff
@@ -0,0 +1,32 @@
+import socket
+import datetime
+
+sock = socket.socket(
+    socket.AF_INET,
+    socket.SOCK_STREAM,
+    socket.IPPROTO_TCP)
+
+address = ('127.0.0.1', 3000)
+
+sock.bind(address)
+
+sock.listen(1)
+
+print(f'--- Starting server on port 3000 at {datetime.datetime.today()} ---')
+
+try:
+    conn, addr = sock.accept()
+
+    while True:
+        data = conn.recv(2048)
+        decoded = data.decode('utf8')
+        if not data:
```

Comment:
This will never happen. The `.recv()` method waits until it gets any data, so if it does not receive anything the server will just hang. A better check would be on the length of `data`, but that will also hang if the received data is a multiple of the buffer length.